### PR TITLE
Using reserve() and const reference objects

### PR DESCRIPTION
**Pull request by GermanAizek**
@HansKristian-Work I think you'll like minor changes.

### CODE REVIEW

File: spirv_glsl.cpp [hunk 1]
```diff
@@ -2587,8 +2587,8 @@ void CompilerGLSL::emit_flattened_io_block_member(const std::string &basename, c
 
 	// Pass in the varying qualifier here so it will appear in the correct declaration order.
 	// Replace member name while emitting it so it encodes both struct name and member name.
-	auto backup_name = get_member_name(parent_type->self, last_index);
-	auto member_name = to_member_name(*parent_type, last_index);
+	const auto &backup_name = get_member_name(parent_type->self, last_index);
+	const auto &member_name = to_member_name(*parent_type, last_index);
```
Comment (HansKristian-Work) [inline spirv_glsl.cpp:2591]:
```suggestion
	auto member_name = to_member_name(*parent_type, last_index);
```

File: spirv_glsl.cpp [hunk 2]
```diff
@@ -2587,7 +2587,7 @@ void CompilerGLSL::emit_flattened_io_block_member(const std::string &basename, c
 
 	// Pass in the varying qualifier here so it will appear in the correct declaration order.
 	// Replace member name while emitting it so it encodes both struct name and member name.
-	auto backup_name = get_member_name(parent_type->self, last_index);
+	const auto &backup_name = get_member_name(parent_type->self, last_index);
```
Comment (HansKristian-Work) [inline spirv_glsl.cpp:2590]:
```suggestion
```

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
get_member_name here must be a value, not reference. It's used after set_member_name as a backup name, and tests will fail here. The reserve is fine, but that's the only thing that's fine after local testing. Please squash the PR into just that.

**Comment by GermanAizek** [comment]
> get_member_name here must be a value, not reference. It's used after set_member_name as a backup name, and tests will fail here. The reserve is fine, but that's the only thing that's fine after local testing. Please squash the PR into just that.

well done.

File: spirv_hlsl.cpp [hunk 3]
```diff
@@ -2734,7 +2734,7 @@ void CompilerHLSL::emit_buffer_block(const SPIRVariable &var)
 			for (auto &member : type.member_types)
 			{
 				add_member_name(type, i);
-				auto backup_name = get_member_name(type.self, i);
+				const auto &backup_name = get_member_name(type.self, i);
```
Comment (HansKristian-Work) [inline spirv_hlsl.cpp:2737]: This is still broken.

File: spirv_hlsl.cpp [hunk 4]
```diff
@@ -2821,7 +2821,7 @@ void CompilerHLSL::emit_push_constant_block(const SPIRVariable &var)
 					const auto &member = type.member_types[i];
 
 					add_member_name(type, constant_index);
-					auto backup_name = get_member_name(type.self, i);
+					const auto &backup_name = get_member_name(type.self, i);
```
Comment (HansKristian-Work) [inline spirv_hlsl.cpp:2824]: Still broken.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
comments

File: spirv_hlsl.cpp (hunk 4 above)
Comment (GermanAizek) [reply spirv_hlsl.cpp]: fix it https://github.com/KhronosGroup/SPIRV-Cross/pull/2245/commits/707f8c6e8395cb319c1abfd301d1bbed02637998

File: spirv_hlsl.cpp (hunk 3 above)
Comment (GermanAizek) [reply spirv_hlsl.cpp]: fix it https://github.com/KhronosGroup/SPIRV-Cross/pull/2245/commits/707f8c6e8395cb319c1abfd301d1bbed02637998

**Comment by HansKristian-Work** [comment]
It's still wrong to use a const-ref here. I just pushed the reserve to main, it's trivial.

**Comment by GermanAizek** [comment]
Ok thx for cherry-pick commit.

**Pull request closed without merging**